// @aws-sdk/client-cloudfront/dist-es/commands/CreateCloudFrontOriginAccessIdentityCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateCloudFrontOriginAccessIdentityRequestFilterSensitiveLog, CreateCloudFrontOriginAccessIdentityResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restXmlCreateCloudFrontOriginAccessIdentityCommand, serializeAws_restXmlCreateCloudFrontOriginAccessIdentityCommand, } from "../protocols/Aws_restXml";
export class CreateCloudFrontOriginAccessIdentityCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, CreateCloudFrontOriginAccessIdentityCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "CloudFrontClient";
        const commandName = "CreateCloudFrontOriginAccessIdentityCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: CreateCloudFrontOriginAccessIdentityRequestFilterSensitiveLog,
            outputFilterSensitiveLog: CreateCloudFrontOriginAccessIdentityResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_restXmlCreateCloudFrontOriginAccessIdentityCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_restXmlCreateCloudFrontOriginAccessIdentityCommand(output, context);
    }
}
