// @aws-sdk/client-cloudfront/dist-es/commands/ListCachePoliciesCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListCachePoliciesRequestFilterSensitiveLog, ListCachePoliciesResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_restXmlListCachePoliciesCommand, serializeAws_restXmlListCachePoliciesCommand, } from "../protocols/Aws_restXml";
export class ListCachePoliciesCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, ListCachePoliciesCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "CloudFrontClient";
        const commandName = "ListCachePoliciesCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListCachePoliciesRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListCachePoliciesResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_restXmlListCachePoliciesCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_restXmlListCachePoliciesCommand(output, context);
    }
}
