// @aws-sdk/client-cloudfront/dist-es/commands/ListFieldLevelEncryptionProfilesCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListFieldLevelEncryptionProfilesRequestFilterSensitiveLog, ListFieldLevelEncryptionProfilesResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_restXmlListFieldLevelEncryptionProfilesCommand, serializeAws_restXmlListFieldLevelEncryptionProfilesCommand, } from "../protocols/Aws_restXml";
export class ListFieldLevelEncryptionProfilesCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, ListFieldLevelEncryptionProfilesCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "CloudFrontClient";
        const commandName = "ListFieldLevelEncryptionProfilesCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListFieldLevelEncryptionProfilesRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListFieldLevelEncryptionProfilesResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_restXmlListFieldLevelEncryptionProfilesCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_restXmlListFieldLevelEncryptionProfilesCommand(output, context);
    }
}
