// @aws-sdk/client-cloudfront/dist-es/models/models_0.js
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
import { CloudFrontServiceException as __BaseException } from "./CloudFrontServiceException";
export var ResponseHeadersPolicyAccessControlAllowMethodsValues;
(function (ResponseHeadersPolicyAccessControlAllowMethodsValues) {
    ResponseHeadersPolicyAccessControlAllowMethodsValues["ALL"] = "ALL";
    ResponseHeadersPolicyAccessControlAllowMethodsValues["DELETE"] = "DELETE";
    ResponseHeadersPolicyAccessControlAllowMethodsValues["GET"] = "GET";
    ResponseHeadersPolicyAccessControlAllowMethodsValues["HEAD"] = "HEAD";
    ResponseHeadersPolicyAccessControlAllowMethodsValues["OPTIONS"] = "OPTIONS";
    ResponseHeadersPolicyAccessControlAllowMethodsValues["PATCH"] = "PATCH";
    ResponseHeadersPolicyAccessControlAllowMethodsValues["POST"] = "POST";
    ResponseHeadersPolicyAccessControlAllowMethodsValues["PUT"] = "PUT";
})(ResponseHeadersPolicyAccessControlAllowMethodsValues || (ResponseHeadersPolicyAccessControlAllowMethodsValues = {}));
export class AccessDenied extends __BaseException {
    constructor(opts) {
        super({
            name: "AccessDenied",
            $fault: "client",
            ...opts,
        });
        this.name = "AccessDenied";
        this.$fault = "client";
        Object.setPrototypeOf(this, AccessDenied.prototype);
        this.Message = opts.Message;
    }
}
export var ICPRecordalStatus;
(function (ICPRecordalStatus) {
    ICPRecordalStatus["APPROVED"] = "APPROVED";
    ICPRecordalStatus["PENDING"] = "PENDING";
    ICPRecordalStatus["SUSPENDED"] = "SUSPENDED";
})(ICPRecordalStatus || (ICPRecordalStatus = {}));
export var Method;
(function (Method) {
    Method["DELETE"] = "DELETE";
    Method["GET"] = "GET";
    Method["HEAD"] = "HEAD";
    Method["OPTIONS"] = "OPTIONS";
    Method["PATCH"] = "PATCH";
    Method["POST"] = "POST";
    Method["PUT"] = "PUT";
})(Method || (Method = {}));
export class IllegalUpdate extends __BaseException {
    constructor(opts) {
        super({
            name: "IllegalUpdate",
            $fault: "client",
            ...opts,
        });
        this.name = "IllegalUpdate";
        this.$fault = "client";
        Object.setPrototypeOf(this, IllegalUpdate.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidArgument extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidArgument",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidArgument";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidArgument.prototype);
        this.Message = opts.Message;
    }
}
export class NoSuchDistribution extends __BaseException {
    constructor(opts) {
        super({
            name: "NoSuchDistribution",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchDistribution";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchDistribution.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyDistributionCNAMEs extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyDistributionCNAMEs",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionCNAMEs";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionCNAMEs.prototype);
        this.Message = opts.Message;
    }
}
export class BatchTooLarge extends __BaseException {
    constructor(opts) {
        super({
            name: "BatchTooLarge",
            $fault: "client",
            ...opts,
        });
        this.name = "BatchTooLarge";
        this.$fault = "client";
        Object.setPrototypeOf(this, BatchTooLarge.prototype);
        this.Message = opts.Message;
    }
}
export var ItemSelection;
(function (ItemSelection) {
    ItemSelection["all"] = "all";
    ItemSelection["none"] = "none";
    ItemSelection["whitelist"] = "whitelist";
})(ItemSelection || (ItemSelection = {}));
export var EventType;
(function (EventType) {
    EventType["origin_request"] = "origin-request";
    EventType["origin_response"] = "origin-response";
    EventType["viewer_request"] = "viewer-request";
    EventType["viewer_response"] = "viewer-response";
})(EventType || (EventType = {}));
export var ViewerProtocolPolicy;
(function (ViewerProtocolPolicy) {
    ViewerProtocolPolicy["allow_all"] = "allow-all";
    ViewerProtocolPolicy["https_only"] = "https-only";
    ViewerProtocolPolicy["redirect_to_https"] = "redirect-to-https";
})(ViewerProtocolPolicy || (ViewerProtocolPolicy = {}));
export var CachePolicyCookieBehavior;
(function (CachePolicyCookieBehavior) {
    CachePolicyCookieBehavior["all"] = "all";
    CachePolicyCookieBehavior["allExcept"] = "allExcept";
    CachePolicyCookieBehavior["none"] = "none";
    CachePolicyCookieBehavior["whitelist"] = "whitelist";
})(CachePolicyCookieBehavior || (CachePolicyCookieBehavior = {}));
export var CachePolicyHeaderBehavior;
(function (CachePolicyHeaderBehavior) {
    CachePolicyHeaderBehavior["none"] = "none";
    CachePolicyHeaderBehavior["whitelist"] = "whitelist";
})(CachePolicyHeaderBehavior || (CachePolicyHeaderBehavior = {}));
export var CachePolicyQueryStringBehavior;
(function (CachePolicyQueryStringBehavior) {
    CachePolicyQueryStringBehavior["all"] = "all";
    CachePolicyQueryStringBehavior["allExcept"] = "allExcept";
    CachePolicyQueryStringBehavior["none"] = "none";
    CachePolicyQueryStringBehavior["whitelist"] = "whitelist";
})(CachePolicyQueryStringBehavior || (CachePolicyQueryStringBehavior = {}));
export class CachePolicyAlreadyExists extends __BaseException {
    constructor(opts) {
        super({
            name: "CachePolicyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "CachePolicyAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, CachePolicyAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
export class CachePolicyInUse extends __BaseException {
    constructor(opts) {
        super({
            name: "CachePolicyInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "CachePolicyInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, CachePolicyInUse.prototype);
        this.Message = opts.Message;
    }
}
export var CachePolicyType;
(function (CachePolicyType) {
    CachePolicyType["custom"] = "custom";
    CachePolicyType["managed"] = "managed";
})(CachePolicyType || (CachePolicyType = {}));
export class CannotChangeImmutablePublicKeyFields extends __BaseException {
    constructor(opts) {
        super({
            name: "CannotChangeImmutablePublicKeyFields",
            $fault: "client",
            ...opts,
        });
        this.name = "CannotChangeImmutablePublicKeyFields";
        this.$fault = "client";
        Object.setPrototypeOf(this, CannotChangeImmutablePublicKeyFields.prototype);
        this.Message = opts.Message;
    }
}
export var CertificateSource;
(function (CertificateSource) {
    CertificateSource["acm"] = "acm";
    CertificateSource["cloudfront"] = "cloudfront";
    CertificateSource["iam"] = "iam";
})(CertificateSource || (CertificateSource = {}));
export class CNAMEAlreadyExists extends __BaseException {
    constructor(opts) {
        super({
            name: "CNAMEAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "CNAMEAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, CNAMEAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
export var HttpVersion;
(function (HttpVersion) {
    HttpVersion["http1_1"] = "http1.1";
    HttpVersion["http2"] = "http2";
    HttpVersion["http2and3"] = "http2and3";
    HttpVersion["http3"] = "http3";
})(HttpVersion || (HttpVersion = {}));
export var OriginProtocolPolicy;
(function (OriginProtocolPolicy) {
    OriginProtocolPolicy["http_only"] = "http-only";
    OriginProtocolPolicy["https_only"] = "https-only";
    OriginProtocolPolicy["match_viewer"] = "match-viewer";
})(OriginProtocolPolicy || (OriginProtocolPolicy = {}));
export var SslProtocol;
(function (SslProtocol) {
    SslProtocol["SSLv3"] = "SSLv3";
    SslProtocol["TLSv1"] = "TLSv1";
    SslProtocol["TLSv1_1"] = "TLSv1.1";
    SslProtocol["TLSv1_2"] = "TLSv1.2";
})(SslProtocol || (SslProtocol = {}));
export var PriceClass;
(function (PriceClass) {
    PriceClass["PriceClass_100"] = "PriceClass_100";
    PriceClass["PriceClass_200"] = "PriceClass_200";
    PriceClass["PriceClass_All"] = "PriceClass_All";
})(PriceClass || (PriceClass = {}));
export var GeoRestrictionType;
(function (GeoRestrictionType) {
    GeoRestrictionType["blacklist"] = "blacklist";
    GeoRestrictionType["none"] = "none";
    GeoRestrictionType["whitelist"] = "whitelist";
})(GeoRestrictionType || (GeoRestrictionType = {}));
export var MinimumProtocolVersion;
(function (MinimumProtocolVersion) {
    MinimumProtocolVersion["SSLv3"] = "SSLv3";
    MinimumProtocolVersion["TLSv1"] = "TLSv1";
    MinimumProtocolVersion["TLSv1_1_2016"] = "TLSv1.1_2016";
    MinimumProtocolVersion["TLSv1_2016"] = "TLSv1_2016";
    MinimumProtocolVersion["TLSv1_2_2018"] = "TLSv1.2_2018";
    MinimumProtocolVersion["TLSv1_2_2019"] = "TLSv1.2_2019";
    MinimumProtocolVersion["TLSv1_2_2021"] = "TLSv1.2_2021";
})(MinimumProtocolVersion || (MinimumProtocolVersion = {}));
export var SSLSupportMethod;
(function (SSLSupportMethod) {
    SSLSupportMethod["sni_only"] = "sni-only";
    SSLSupportMethod["static_ip"] = "static-ip";
    SSLSupportMethod["vip"] = "vip";
})(SSLSupportMethod || (SSLSupportMethod = {}));
export class DistributionAlreadyExists extends __BaseException {
    constructor(opts) {
        super({
            name: "DistributionAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "DistributionAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, DistributionAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
export class IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior extends __BaseException {
    constructor(opts) {
        super({
            name: "IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior",
            $fault: "client",
            ...opts,
        });
        this.name = "IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior";
        this.$fault = "client";
        Object.setPrototypeOf(this, IllegalFieldLevelEncryptionConfigAssociationWithCacheBehavior.prototype);
        this.Message = opts.Message;
    }
}
export class InconsistentQuantities extends __BaseException {
    constructor(opts) {
        super({
            name: "InconsistentQuantities",
            $fault: "client",
            ...opts,
        });
        this.name = "InconsistentQuantities";
        this.$fault = "client";
        Object.setPrototypeOf(this, InconsistentQuantities.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidDefaultRootObject extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidDefaultRootObject",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidDefaultRootObject";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidDefaultRootObject.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidErrorCode extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidErrorCode",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidErrorCode";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidErrorCode.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidForwardCookies extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidForwardCookies",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidForwardCookies";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidForwardCookies.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidFunctionAssociation extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidFunctionAssociation",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidFunctionAssociation";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidFunctionAssociation.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidGeoRestrictionParameter extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidGeoRestrictionParameter",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidGeoRestrictionParameter";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidGeoRestrictionParameter.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidHeadersForS3Origin extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidHeadersForS3Origin",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidHeadersForS3Origin";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidHeadersForS3Origin.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidIfMatchVersion extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidIfMatchVersion",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidIfMatchVersion";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidIfMatchVersion.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidLambdaFunctionAssociation extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidLambdaFunctionAssociation",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidLambdaFunctionAssociation";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidLambdaFunctionAssociation.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidLocationCode extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidLocationCode",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidLocationCode";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidLocationCode.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidMinimumProtocolVersion extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidMinimumProtocolVersion",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidMinimumProtocolVersion";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidMinimumProtocolVersion.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidOrigin extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidOrigin",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidOrigin";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidOrigin.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidOriginAccessControl extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidOriginAccessControl",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidOriginAccessControl";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidOriginAccessControl.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidOriginAccessIdentity extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidOriginAccessIdentity",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidOriginAccessIdentity";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidOriginAccessIdentity.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidOriginKeepaliveTimeout extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidOriginKeepaliveTimeout",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidOriginKeepaliveTimeout";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidOriginKeepaliveTimeout.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidOriginReadTimeout extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidOriginReadTimeout",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidOriginReadTimeout";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidOriginReadTimeout.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidProtocolSettings extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidProtocolSettings",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidProtocolSettings";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidProtocolSettings.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidQueryStringParameters extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidQueryStringParameters",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidQueryStringParameters";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidQueryStringParameters.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidRelativePath extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidRelativePath",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidRelativePath";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidRelativePath.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidRequiredProtocol extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidRequiredProtocol",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidRequiredProtocol";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidRequiredProtocol.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidResponseCode extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidResponseCode",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidResponseCode";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidResponseCode.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidTTLOrder extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidTTLOrder",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidTTLOrder";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidTTLOrder.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidViewerCertificate extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidViewerCertificate",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidViewerCertificate";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidViewerCertificate.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidWebACLId extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidWebACLId",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidWebACLId";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidWebACLId.prototype);
        this.Message = opts.Message;
    }
}
export class MissingBody extends __BaseException {
    constructor(opts) {
        super({
            name: "MissingBody",
            $fault: "client",
            ...opts,
        });
        this.name = "MissingBody";
        this.$fault = "client";
        Object.setPrototypeOf(this, MissingBody.prototype);
        this.Message = opts.Message;
    }
}
export class NoSuchCachePolicy extends __BaseException {
    constructor(opts) {
        super({
            name: "NoSuchCachePolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchCachePolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchCachePolicy.prototype);
        this.Message = opts.Message;
    }
}
export class NoSuchFieldLevelEncryptionConfig extends __BaseException {
    constructor(opts) {
        super({
            name: "NoSuchFieldLevelEncryptionConfig",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchFieldLevelEncryptionConfig";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchFieldLevelEncryptionConfig.prototype);
        this.Message = opts.Message;
    }
}
export class NoSuchOrigin extends __BaseException {
    constructor(opts) {
        super({
            name: "NoSuchOrigin",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchOrigin";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchOrigin.prototype);
        this.Message = opts.Message;
    }
}
export class NoSuchOriginRequestPolicy extends __BaseException {
    constructor(opts) {
        super({
            name: "NoSuchOriginRequestPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchOriginRequestPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchOriginRequestPolicy.prototype);
        this.Message = opts.Message;
    }
}
export class NoSuchRealtimeLogConfig extends __BaseException {
    constructor(opts) {
        super({
            name: "NoSuchRealtimeLogConfig",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchRealtimeLogConfig";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchRealtimeLogConfig.prototype);
        this.Message = opts.Message;
    }
}
export class NoSuchResponseHeadersPolicy extends __BaseException {
    constructor(opts) {
        super({
            name: "NoSuchResponseHeadersPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchResponseHeadersPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchResponseHeadersPolicy.prototype);
        this.Message = opts.Message;
    }
}
export class PreconditionFailed extends __BaseException {
    constructor(opts) {
        super({
            name: "PreconditionFailed",
            $fault: "client",
            ...opts,
        });
        this.name = "PreconditionFailed";
        this.$fault = "client";
        Object.setPrototypeOf(this, PreconditionFailed.prototype);
        this.Message = opts.Message;
    }
}
export class RealtimeLogConfigOwnerMismatch extends __BaseException {
    constructor(opts) {
        super({
            name: "RealtimeLogConfigOwnerMismatch",
            $fault: "client",
            ...opts,
        });
        this.name = "RealtimeLogConfigOwnerMismatch";
        this.$fault = "client";
        Object.setPrototypeOf(this, RealtimeLogConfigOwnerMismatch.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyCacheBehaviors extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyCacheBehaviors",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCacheBehaviors";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCacheBehaviors.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyCertificates extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyCertificates",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCertificates";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCertificates.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyCookieNamesInWhiteList extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyCookieNamesInWhiteList",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCookieNamesInWhiteList";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCookieNamesInWhiteList.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyDistributions extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyDistributions",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributions";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributions.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyDistributionsAssociatedToCachePolicy extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToCachePolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsAssociatedToCachePolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToCachePolicy.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyDistributionsAssociatedToFieldLevelEncryptionConfig extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToFieldLevelEncryptionConfig",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsAssociatedToFieldLevelEncryptionConfig";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToFieldLevelEncryptionConfig.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyDistributionsAssociatedToKeyGroup extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToKeyGroup",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsAssociatedToKeyGroup";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToKeyGroup.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyDistributionsAssociatedToOriginRequestPolicy extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToOriginRequestPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsAssociatedToOriginRequestPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToOriginRequestPolicy.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyDistributionsAssociatedToResponseHeadersPolicy extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToResponseHeadersPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsAssociatedToResponseHeadersPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToResponseHeadersPolicy.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyDistributionsWithFunctionAssociations extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsWithFunctionAssociations",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsWithFunctionAssociations";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsWithFunctionAssociations.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyDistributionsWithLambdaAssociations extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsWithLambdaAssociations",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsWithLambdaAssociations";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsWithLambdaAssociations.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyDistributionsWithSingleFunctionARN extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsWithSingleFunctionARN",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsWithSingleFunctionARN";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsWithSingleFunctionARN.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyFunctionAssociations extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyFunctionAssociations",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFunctionAssociations";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFunctionAssociations.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyHeadersInForwardedValues extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyHeadersInForwardedValues",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyHeadersInForwardedValues";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyHeadersInForwardedValues.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyKeyGroupsAssociatedToDistribution extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyKeyGroupsAssociatedToDistribution",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyKeyGroupsAssociatedToDistribution";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyKeyGroupsAssociatedToDistribution.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyLambdaFunctionAssociations extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyLambdaFunctionAssociations",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyLambdaFunctionAssociations";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyLambdaFunctionAssociations.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyOriginCustomHeaders extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyOriginCustomHeaders",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyOriginCustomHeaders";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyOriginCustomHeaders.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyOriginGroupsPerDistribution extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyOriginGroupsPerDistribution",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyOriginGroupsPerDistribution";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyOriginGroupsPerDistribution.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyOrigins extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyOrigins",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyOrigins";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyOrigins.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyQueryStringParameters extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyQueryStringParameters",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyQueryStringParameters";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyQueryStringParameters.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyTrustedSigners extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyTrustedSigners",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyTrustedSigners";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyTrustedSigners.prototype);
        this.Message = opts.Message;
    }
}
export class TrustedKeyGroupDoesNotExist extends __BaseException {
    constructor(opts) {
        super({
            name: "TrustedKeyGroupDoesNotExist",
            $fault: "client",
            ...opts,
        });
        this.name = "TrustedKeyGroupDoesNotExist";
        this.$fault = "client";
        Object.setPrototypeOf(this, TrustedKeyGroupDoesNotExist.prototype);
        this.Message = opts.Message;
    }
}
export class TrustedSignerDoesNotExist extends __BaseException {
    constructor(opts) {
        super({
            name: "TrustedSignerDoesNotExist",
            $fault: "client",
            ...opts,
        });
        this.name = "TrustedSignerDoesNotExist";
        this.$fault = "client";
        Object.setPrototypeOf(this, TrustedSignerDoesNotExist.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyCachePolicies extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyCachePolicies",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCachePolicies";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCachePolicies.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyCookiesInCachePolicy extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyCookiesInCachePolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCookiesInCachePolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCookiesInCachePolicy.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyHeadersInCachePolicy extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyHeadersInCachePolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyHeadersInCachePolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyHeadersInCachePolicy.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyQueryStringsInCachePolicy extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyQueryStringsInCachePolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyQueryStringsInCachePolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyQueryStringsInCachePolicy.prototype);
        this.Message = opts.Message;
    }
}
export class CloudFrontOriginAccessIdentityAlreadyExists extends __BaseException {
    constructor(opts) {
        super({
            name: "CloudFrontOriginAccessIdentityAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "CloudFrontOriginAccessIdentityAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, CloudFrontOriginAccessIdentityAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyCloudFrontOriginAccessIdentities extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyCloudFrontOriginAccessIdentities",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCloudFrontOriginAccessIdentities";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCloudFrontOriginAccessIdentities.prototype);
        this.Message = opts.Message;
    }
}
export var ContinuousDeploymentPolicyType;
(function (ContinuousDeploymentPolicyType) {
    ContinuousDeploymentPolicyType["SingleHeader"] = "SingleHeader";
    ContinuousDeploymentPolicyType["SingleWeight"] = "SingleWeight";
})(ContinuousDeploymentPolicyType || (ContinuousDeploymentPolicyType = {}));
export class StagingDistributionInUse extends __BaseException {
    constructor(opts) {
        super({
            name: "StagingDistributionInUse",
            $fault: "client",
            ...opts,
        });
        this.name = "StagingDistributionInUse";
        this.$fault = "client";
        Object.setPrototypeOf(this, StagingDistributionInUse.prototype);
        this.Message = opts.Message;
    }
}
export class IllegalOriginAccessConfiguration extends __BaseException {
    constructor(opts) {
        super({
            name: "IllegalOriginAccessConfiguration",
            $fault: "client",
            ...opts,
        });
        this.name = "IllegalOriginAccessConfiguration";
        this.$fault = "client";
        Object.setPrototypeOf(this, IllegalOriginAccessConfiguration.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidDomainNameForOriginAccessControl extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidDomainNameForOriginAccessControl",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidDomainNameForOriginAccessControl";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidDomainNameForOriginAccessControl.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyDistributionsAssociatedToOriginAccessControl extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyDistributionsAssociatedToOriginAccessControl",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyDistributionsAssociatedToOriginAccessControl";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyDistributionsAssociatedToOriginAccessControl.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidTagging extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidTagging",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidTagging";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidTagging.prototype);
        this.Message = opts.Message;
    }
}
export var Format;
(function (Format) {
    Format["URLEncoded"] = "URLEncoded";
})(Format || (Format = {}));
export class FieldLevelEncryptionConfigAlreadyExists extends __BaseException {
    constructor(opts) {
        super({
            name: "FieldLevelEncryptionConfigAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "FieldLevelEncryptionConfigAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, FieldLevelEncryptionConfigAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
export class NoSuchFieldLevelEncryptionProfile extends __BaseException {
    constructor(opts) {
        super({
            name: "NoSuchFieldLevelEncryptionProfile",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchFieldLevelEncryptionProfile";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchFieldLevelEncryptionProfile.prototype);
        this.Message = opts.Message;
    }
}
export class QueryArgProfileEmpty extends __BaseException {
    constructor(opts) {
        super({
            name: "QueryArgProfileEmpty",
            $fault: "client",
            ...opts,
        });
        this.name = "QueryArgProfileEmpty";
        this.$fault = "client";
        Object.setPrototypeOf(this, QueryArgProfileEmpty.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyFieldLevelEncryptionConfigs extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionConfigs",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFieldLevelEncryptionConfigs";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionConfigs.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyFieldLevelEncryptionContentTypeProfiles extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionContentTypeProfiles",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFieldLevelEncryptionContentTypeProfiles";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionContentTypeProfiles.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyFieldLevelEncryptionQueryArgProfiles extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionQueryArgProfiles",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFieldLevelEncryptionQueryArgProfiles";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionQueryArgProfiles.prototype);
        this.Message = opts.Message;
    }
}
export class FieldLevelEncryptionProfileAlreadyExists extends __BaseException {
    constructor(opts) {
        super({
            name: "FieldLevelEncryptionProfileAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "FieldLevelEncryptionProfileAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, FieldLevelEncryptionProfileAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
export class FieldLevelEncryptionProfileSizeExceeded extends __BaseException {
    constructor(opts) {
        super({
            name: "FieldLevelEncryptionProfileSizeExceeded",
            $fault: "client",
            ...opts,
        });
        this.name = "FieldLevelEncryptionProfileSizeExceeded";
        this.$fault = "client";
        Object.setPrototypeOf(this, FieldLevelEncryptionProfileSizeExceeded.prototype);
        this.Message = opts.Message;
    }
}
export class NoSuchPublicKey extends __BaseException {
    constructor(opts) {
        super({
            name: "NoSuchPublicKey",
            $fault: "client",
            ...opts,
        });
        this.name = "NoSuchPublicKey";
        this.$fault = "client";
        Object.setPrototypeOf(this, NoSuchPublicKey.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyFieldLevelEncryptionEncryptionEntities extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionEncryptionEntities",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFieldLevelEncryptionEncryptionEntities";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionEncryptionEntities.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyFieldLevelEncryptionFieldPatterns extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionFieldPatterns",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFieldLevelEncryptionFieldPatterns";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionFieldPatterns.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyFieldLevelEncryptionProfiles extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyFieldLevelEncryptionProfiles",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFieldLevelEncryptionProfiles";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFieldLevelEncryptionProfiles.prototype);
        this.Message = opts.Message;
    }
}
export var FunctionRuntime;
(function (FunctionRuntime) {
    FunctionRuntime["cloudfront_js_1_0"] = "cloudfront-js-1.0";
})(FunctionRuntime || (FunctionRuntime = {}));
export var FunctionStage;
(function (FunctionStage) {
    FunctionStage["DEVELOPMENT"] = "DEVELOPMENT";
    FunctionStage["LIVE"] = "LIVE";
})(FunctionStage || (FunctionStage = {}));
export class FunctionAlreadyExists extends __BaseException {
    constructor(opts) {
        super({
            name: "FunctionAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "FunctionAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, FunctionAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
export class FunctionSizeLimitExceeded extends __BaseException {
    constructor(opts) {
        super({
            name: "FunctionSizeLimitExceeded",
            $fault: "client",
            ...opts,
        });
        this.name = "FunctionSizeLimitExceeded";
        this.$fault = "client";
        Object.setPrototypeOf(this, FunctionSizeLimitExceeded.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyFunctions extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyFunctions",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyFunctions";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyFunctions.prototype);
        this.Message = opts.Message;
    }
}
export class UnsupportedOperation extends __BaseException {
    constructor(opts) {
        super({
            name: "UnsupportedOperation",
            $fault: "client",
            ...opts,
        });
        this.name = "UnsupportedOperation";
        this.$fault = "client";
        Object.setPrototypeOf(this, UnsupportedOperation.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyInvalidationsInProgress extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyInvalidationsInProgress",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyInvalidationsInProgress";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyInvalidationsInProgress.prototype);
        this.Message = opts.Message;
    }
}
export class KeyGroupAlreadyExists extends __BaseException {
    constructor(opts) {
        super({
            name: "KeyGroupAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "KeyGroupAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, KeyGroupAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyKeyGroups extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyKeyGroups",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyKeyGroups";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyKeyGroups.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyPublicKeysInKeyGroup extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyPublicKeysInKeyGroup",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyPublicKeysInKeyGroup";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyPublicKeysInKeyGroup.prototype);
        this.Message = opts.Message;
    }
}
export var RealtimeMetricsSubscriptionStatus;
(function (RealtimeMetricsSubscriptionStatus) {
    RealtimeMetricsSubscriptionStatus["Disabled"] = "Disabled";
    RealtimeMetricsSubscriptionStatus["Enabled"] = "Enabled";
})(RealtimeMetricsSubscriptionStatus || (RealtimeMetricsSubscriptionStatus = {}));
export class MonitoringSubscriptionAlreadyExists extends __BaseException {
    constructor(opts) {
        super({
            name: "MonitoringSubscriptionAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "MonitoringSubscriptionAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, MonitoringSubscriptionAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
export var OriginAccessControlOriginTypes;
(function (OriginAccessControlOriginTypes) {
    OriginAccessControlOriginTypes["s3"] = "s3";
})(OriginAccessControlOriginTypes || (OriginAccessControlOriginTypes = {}));
export var OriginAccessControlSigningBehaviors;
(function (OriginAccessControlSigningBehaviors) {
    OriginAccessControlSigningBehaviors["always"] = "always";
    OriginAccessControlSigningBehaviors["never"] = "never";
    OriginAccessControlSigningBehaviors["no_override"] = "no-override";
})(OriginAccessControlSigningBehaviors || (OriginAccessControlSigningBehaviors = {}));
export var OriginAccessControlSigningProtocols;
(function (OriginAccessControlSigningProtocols) {
    OriginAccessControlSigningProtocols["sigv4"] = "sigv4";
})(OriginAccessControlSigningProtocols || (OriginAccessControlSigningProtocols = {}));
export class OriginAccessControlAlreadyExists extends __BaseException {
    constructor(opts) {
        super({
            name: "OriginAccessControlAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "OriginAccessControlAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, OriginAccessControlAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyOriginAccessControls extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyOriginAccessControls",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyOriginAccessControls";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyOriginAccessControls.prototype);
        this.Message = opts.Message;
    }
}
export var OriginRequestPolicyCookieBehavior;
(function (OriginRequestPolicyCookieBehavior) {
    OriginRequestPolicyCookieBehavior["all"] = "all";
    OriginRequestPolicyCookieBehavior["none"] = "none";
    OriginRequestPolicyCookieBehavior["whitelist"] = "whitelist";
})(OriginRequestPolicyCookieBehavior || (OriginRequestPolicyCookieBehavior = {}));
export var OriginRequestPolicyHeaderBehavior;
(function (OriginRequestPolicyHeaderBehavior) {
    OriginRequestPolicyHeaderBehavior["allViewer"] = "allViewer";
    OriginRequestPolicyHeaderBehavior["allViewerAndWhitelistCloudFront"] = "allViewerAndWhitelistCloudFront";
    OriginRequestPolicyHeaderBehavior["none"] = "none";
    OriginRequestPolicyHeaderBehavior["whitelist"] = "whitelist";
})(OriginRequestPolicyHeaderBehavior || (OriginRequestPolicyHeaderBehavior = {}));
export var OriginRequestPolicyQueryStringBehavior;
(function (OriginRequestPolicyQueryStringBehavior) {
    OriginRequestPolicyQueryStringBehavior["all"] = "all";
    OriginRequestPolicyQueryStringBehavior["none"] = "none";
    OriginRequestPolicyQueryStringBehavior["whitelist"] = "whitelist";
})(OriginRequestPolicyQueryStringBehavior || (OriginRequestPolicyQueryStringBehavior = {}));
export class OriginRequestPolicyAlreadyExists extends __BaseException {
    constructor(opts) {
        super({
            name: "OriginRequestPolicyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "OriginRequestPolicyAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, OriginRequestPolicyAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyCookiesInOriginRequestPolicy extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyCookiesInOriginRequestPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCookiesInOriginRequestPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCookiesInOriginRequestPolicy.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyHeadersInOriginRequestPolicy extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyHeadersInOriginRequestPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyHeadersInOriginRequestPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyHeadersInOriginRequestPolicy.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyOriginRequestPolicies extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyOriginRequestPolicies",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyOriginRequestPolicies";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyOriginRequestPolicies.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyQueryStringsInOriginRequestPolicy extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyQueryStringsInOriginRequestPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyQueryStringsInOriginRequestPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyQueryStringsInOriginRequestPolicy.prototype);
        this.Message = opts.Message;
    }
}
export class PublicKeyAlreadyExists extends __BaseException {
    constructor(opts) {
        super({
            name: "PublicKeyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "PublicKeyAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, PublicKeyAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyPublicKeys extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyPublicKeys",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyPublicKeys";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyPublicKeys.prototype);
        this.Message = opts.Message;
    }
}
export class RealtimeLogConfigAlreadyExists extends __BaseException {
    constructor(opts) {
        super({
            name: "RealtimeLogConfigAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "RealtimeLogConfigAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, RealtimeLogConfigAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyRealtimeLogConfigs extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyRealtimeLogConfigs",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyRealtimeLogConfigs";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyRealtimeLogConfigs.prototype);
        this.Message = opts.Message;
    }
}
export var FrameOptionsList;
(function (FrameOptionsList) {
    FrameOptionsList["DENY"] = "DENY";
    FrameOptionsList["SAMEORIGIN"] = "SAMEORIGIN";
})(FrameOptionsList || (FrameOptionsList = {}));
export var ReferrerPolicyList;
(function (ReferrerPolicyList) {
    ReferrerPolicyList["no_referrer"] = "no-referrer";
    ReferrerPolicyList["no_referrer_when_downgrade"] = "no-referrer-when-downgrade";
    ReferrerPolicyList["origin"] = "origin";
    ReferrerPolicyList["origin_when_cross_origin"] = "origin-when-cross-origin";
    ReferrerPolicyList["same_origin"] = "same-origin";
    ReferrerPolicyList["strict_origin"] = "strict-origin";
    ReferrerPolicyList["strict_origin_when_cross_origin"] = "strict-origin-when-cross-origin";
    ReferrerPolicyList["unsafe_url"] = "unsafe-url";
})(ReferrerPolicyList || (ReferrerPolicyList = {}));
export class ResponseHeadersPolicyAlreadyExists extends __BaseException {
    constructor(opts) {
        super({
            name: "ResponseHeadersPolicyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        this.name = "ResponseHeadersPolicyAlreadyExists";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResponseHeadersPolicyAlreadyExists.prototype);
        this.Message = opts.Message;
    }
}
export class TooLongCSPInResponseHeadersPolicy extends __BaseException {
    constructor(opts) {
        super({
            name: "TooLongCSPInResponseHeadersPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooLongCSPInResponseHeadersPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooLongCSPInResponseHeadersPolicy.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyCustomHeadersInResponseHeadersPolicy extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyCustomHeadersInResponseHeadersPolicy",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyCustomHeadersInResponseHeadersPolicy";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyCustomHeadersInResponseHeadersPolicy.prototype);
        this.Message = opts.Message;
    }
}
export class TooManyResponseHeadersPolicies extends __BaseException {
    constructor(opts) {
        super({
            name: "TooManyResponseHeadersPolicies",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyResponseHeadersPolicies";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyResponseHeadersPolicies.prototype);
        this.Message = opts.Message;
    }
}
export const KeyPairIdsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const KGKeyPairIdsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ActiveTrustedKeyGroupsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SignerFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ActiveTrustedSignersFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const AliasesFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const AliasICPRecordalFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CachedMethodsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const AllowedMethodsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const AssociateAliasRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CookieNamesFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CookiePreferenceFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const HeadersFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const QueryStringCacheKeysFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ForwardedValuesFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const FunctionAssociationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const FunctionAssociationsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const LambdaFunctionAssociationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const LambdaFunctionAssociationsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TrustedKeyGroupsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TrustedSignersFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CacheBehaviorFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CacheBehaviorsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CachePolicyCookiesConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CachePolicyHeadersConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const QueryStringNamesFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CachePolicyQueryStringsConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ParametersInCacheKeyAndForwardedToOriginFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CachePolicyConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CachePolicyFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CachePolicySummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CachePolicyListFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CopyDistributionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CustomErrorResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CustomErrorResponsesFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DefaultCacheBehaviorFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const LoggingConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const StatusCodesFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OriginGroupFailoverCriteriaFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OriginGroupMemberFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OriginGroupMembersFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OriginGroupFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OriginGroupsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OriginCustomHeaderFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.HeaderValue && { HeaderValue: SENSITIVE_STRING }),
});
export const CustomHeadersFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Items && { Items: obj.Items.map((item) => OriginCustomHeaderFilterSensitiveLog(item)) }),
});
export const OriginSslProtocolsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CustomOriginConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OriginShieldFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const S3OriginConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OriginFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.CustomHeaders && { CustomHeaders: CustomHeadersFilterSensitiveLog(obj.CustomHeaders) }),
});
export const OriginsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const GeoRestrictionFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const RestrictionsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ViewerCertificateFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DistributionConfigFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Comment && { Comment: SENSITIVE_STRING }),
});
export const DistributionFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.DistributionConfig && { DistributionConfig: DistributionConfigFilterSensitiveLog(obj.DistributionConfig) }),
});
export const CopyDistributionResultFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Distribution && { Distribution: DistributionFilterSensitiveLog(obj.Distribution) }),
});
export const CreateCachePolicyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateCachePolicyResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CloudFrontOriginAccessIdentityConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateCloudFrontOriginAccessIdentityRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CloudFrontOriginAccessIdentityFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateCloudFrontOriginAccessIdentityResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const StagingDistributionDnsNamesFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ContinuousDeploymentSingleHeaderConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SessionStickinessConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ContinuousDeploymentSingleWeightConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TrafficConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ContinuousDeploymentPolicyConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateContinuousDeploymentPolicyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ContinuousDeploymentPolicyFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateContinuousDeploymentPolicyResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateDistributionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.DistributionConfig && { DistributionConfig: DistributionConfigFilterSensitiveLog(obj.DistributionConfig) }),
});
export const CreateDistributionResultFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Distribution && { Distribution: DistributionFilterSensitiveLog(obj.Distribution) }),
});
export const TagFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TagsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DistributionConfigWithTagsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.DistributionConfig && { DistributionConfig: DistributionConfigFilterSensitiveLog(obj.DistributionConfig) }),
});
export const CreateDistributionWithTagsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.DistributionConfigWithTags && {
        DistributionConfigWithTags: DistributionConfigWithTagsFilterSensitiveLog(obj.DistributionConfigWithTags),
    }),
});
export const CreateDistributionWithTagsResultFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Distribution && { Distribution: DistributionFilterSensitiveLog(obj.Distribution) }),
});
export const ContentTypeProfileFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ContentTypeProfilesFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ContentTypeProfileConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const QueryArgProfileFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const QueryArgProfilesFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const QueryArgProfileConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const FieldLevelEncryptionConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateFieldLevelEncryptionConfigRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const FieldLevelEncryptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateFieldLevelEncryptionConfigResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const FieldPatternsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const EncryptionEntityFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const EncryptionEntitiesFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const FieldLevelEncryptionProfileConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateFieldLevelEncryptionProfileRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const FieldLevelEncryptionProfileFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateFieldLevelEncryptionProfileResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const FunctionConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateFunctionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.FunctionCode && { FunctionCode: SENSITIVE_STRING }),
});
export const FunctionMetadataFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const FunctionSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateFunctionResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const PathsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const InvalidationBatchFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateInvalidationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const InvalidationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateInvalidationResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const KeyGroupConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateKeyGroupRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const KeyGroupFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateKeyGroupResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const RealtimeMetricsSubscriptionConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const MonitoringSubscriptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateMonitoringSubscriptionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateMonitoringSubscriptionResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OriginAccessControlConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateOriginAccessControlRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OriginAccessControlFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateOriginAccessControlResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OriginRequestPolicyCookiesConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OriginRequestPolicyHeadersConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OriginRequestPolicyQueryStringsConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OriginRequestPolicyConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateOriginRequestPolicyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const OriginRequestPolicyFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateOriginRequestPolicyResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const PublicKeyConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreatePublicKeyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const PublicKeyFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreatePublicKeyResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const KinesisStreamConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const EndPointFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateRealtimeLogConfigRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const RealtimeLogConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateRealtimeLogConfigResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyAccessControlAllowHeadersFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyAccessControlAllowMethodsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyAccessControlAllowOriginsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyAccessControlExposeHeadersFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyCorsConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyCustomHeaderFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyCustomHeadersConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyContentSecurityPolicyFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyContentTypeOptionsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyFrameOptionsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyReferrerPolicyFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyStrictTransportSecurityFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyXSSProtectionFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicySecurityHeadersConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyServerTimingHeadersConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateResponseHeadersPolicyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ResponseHeadersPolicyFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateResponseHeadersPolicyResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const StreamingLoggingConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
